lib/pairmotron/pair_maker.ex

```diff
@@ -58,23 +62,23 @@ defmodule Pairmotron.PairMaker do
       |> Mixer.mixify(week)
       |> Pairer.generate_pairs(pairs)
 
-    results.pairs
-      |> Enum.map(fn(users) -> make_pairs(users, year, week, group_id) end)
-      |> List.flatten
-      |> Enum.map(fn(p) -> Repo.insert! p end)
+    multi = results.pairs
+      |> Enum.reduce(multi, fn(users, acc) ->
+          insert_pair(acc, year, week, group_id, users)
+      end)
 
-    insert_user_pair(results.user_pair)
+    multi = insert_lone_user_pair(multi, results.user_pair)
+    Repo.transaction(multi)
```

Comment:
You're definitely right that the user doesn't really gain much from seeing this fail, but something rubs me the wrong way about this failing with no trace. If this starts failing, I'd want to know about it (because it really shouldn't). Can we at least log in the case of a failure?